### PR TITLE
[Feature] add support for vipnas

### PR DESCRIPTION
**Pull request by LiuYi-Up**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Add the support for ViPNAS in ncnn(ncnn-int8) backend.

## Modification

Added:  
'mmdeploy/codebase/mmpose/models/heads/vipnas_heatmap_simple_head.py'  
'mmdeploy/mmcv/cnn/context_block.py'  
'mmdeploy/mmcv/cnn/hsigmoid.py'  
'mmdeploy/mmcv/cnn/hswish.py'  

Modified:  
'mmdeploy/pytorch/ops/layer_norm.py'

Docs: some information in 'quantization.md' and 'supported_models.md'

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmdeploy/codebase/mmpose/models/heads/vipnas_heatmap_simple_head.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+    Args:
+    x (list[torch.Tensor[N,K,H,W]]): Input features.
+    flip_pairs (None | list[tuple]):
+        Pairs of keypoints which are mirrored.
+
+    Returns:
+        output_heatmap (torch.Tensor): Output heatmaps.
+    """
+    assert flip_pairs is None
```
Comment (tpoisonooo) [inline mmdeploy/codebase/mmpose/models/heads/vipnas_heatmap_simple_head.py:22]: why `flip_pairs` must be none ?

**Review by tpoisonooo — APPROVED** [review]

File: mmdeploy/codebase/mmpose/models/heads/vipnas_heatmap_simple_head.py (hunk 1 above)
Comment (LiuYi-Up) [reply mmdeploy/codebase/mmpose/models/heads/vipnas_heatmap_simple_head.py:22]: MMPose models are tested with `flip_test` explicitly set to `False` in model configs. It won't activate the parameter `flip_pairs`.

**Comment by RunningLeon** [comment]
could add unit test for all new rewritings.

File: mmdeploy/backend/ncnn/wrapper.py [hunk 2]
```diff
@@ -102,6 +102,7 @@ def forward(self, inputs: Dict[str,
                 data = input_tensor[batch_id].contiguous()
                 data = data.detach().cpu().numpy()
                 input_mat = ncnn.Mat(data)
+                input_mat = input_mat.clone()
```
Comment (RunningLeon) [inline mmdeploy/backend/ncnn/wrapper.py:105]: why do we need to clone here?

File: tests/test_mmcv/test_mmcv_cnn.py [hunk 3]
```diff
@@ -30,3 +32,97 @@ def test_multiheadattention_ncnn():
...
+        deploy_cfg=deploy_cfg,
+        run_with_backend=True)
+    if is_backend_output is None:
+        assert rewrite_outputs is not None
+    else:
+        assert torch.allclose(
+            model_outputs, rewrite_outputs[0], rtol=1e-03, atol=1e-05)
+
+
+if __name__ == '__main__':
```
Comment (RunningLeon) [inline tests/test_mmcv/test_mmcv_cnn.py:127]: should remove this main function

File: mmdeploy/backend/ncnn/wrapper.py (hunk 2 above)
Comment (LiuYi-Up) [reply mmdeploy/backend/ncnn/wrapper.py:105]: When I try to test the `hardsigmoid` and `hardswish` ops which support `forward_inplace` in `ncnn`, something wrong happened with the `refcount` of `input` (it seems that an exception occurred to the memory address result a `segmentation fault`). So I clone this here and it works to solve the problem.

File: mmdeploy/pytorch/ops/layer_norm.py [hunk 4]
```diff
@@ -56,6 +56,25 @@ def _layer_norm_ncnn(g, input, normalized_shape, weight, bias, eps,
     """
     weight.setDebugName('layernorm_weight')
     bias.setDebugName('layernorm_bias')
+    [c, h, w] = normalized_shape
```
Comment (RunningLeon) [inline mmdeploy/pytorch/ops/layer_norm.py:59]: if `len(normalized_shape)` could be `4`, then this line may break in this case.

**Review by RunningLeon — APPROVED** [review]
LGTM after resolving left comments @LiuYi-Up
